build/azure-pipelines/linux/product-build-linux.yml

```diff
@@ -124,30 +113,62 @@ steps:
-        -v /var/run/docker.sock:/var/run/docker.sock \
-        -u 1000:1000 \
-        -w /home/builduser/vscode vscodehub.azurecr.io/vscode-linux-build-agent:bionic-$(VSCODE_ARCH) \
-        /bin/bash -c "./build/azure-pipelines/linux/install.sh"
-
-        sudo chown -R $USER:$USER /mnt/vss/_work/1/s
+        ./build/azure-pipelines/linux/install.sh
       env:
         npm_config_arch: $(NPM_ARCH)
+        VSCODE_ARCH: $(VSCODE_ARCH)
         NPM_REGISTRY: "$(NPM_REGISTRY)"
         ELECTRON_SKIP_BINARY_DOWNLOAD: 1
         PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
-        VSCODE_SKIP_NODE_VERSION_CHECK: 1
         GITHUB_TOKEN: "$(github-distro-mixin-password)"
-        VSCODE_HOST_MOUNT: "/mnt/vss/_work/1/s"
-        ${{ if or(eq(parameters.VSCODE_ARCH, 'x64'), eq(parameters.VSCODE_ARCH, 'arm64')) }}:
-          VSCODE_REMOTE_DEPENDENCIES_CONTAINER_NAME: vscodehub.azurecr.io/vscode-linux-build-agent:centos7-devtoolset8-$(VSCODE_ARCH)
       displayName: Install dependencies (non-OSS)
       condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
 
+    - script: |
+        set -e
+
+        TRIPLE="x86_64-linux-gnu"
+        if [ "$VSCODE_ARCH" == "arm64" ]; then
+          TRIPLE="aarch64-linux-gnu"
+        elif [ "$VSCODE_ARCH" == "armhf" ]; then
+          TRIPLE="arm-rpi-linux-gnueabihf"
+        fi
+
+        # Get all files with .node extension from remote/node_modules folder
+        files=$(find remote/node_modules -name "*.node")
+
+        # Check if any file has dependency of GLIBC > 2.28 or GLIBCXX > 3.4.25
+        for file in $files; do
+          glibc_version="2.28"
+          glibcxx_version="3.4.25"
+          while IFS= read -r line; do
+            if [[ $line == *"GLIBC_"* ]]; then
+              version=$(echo "$line" | awk '{print $5}' | tr -d '()')
+              version=${version#*_}
+              if [[ $(printf "%s\n%s" "$version" "$glibc_version" | sort -V | tail -n1) == "$version" ]]; then
+                glibc_version=$version
+              fi
+            elif [[ $line == *"GLIBCXX_"* ]]; then
+              version=$(echo "$line" | awk '{print $5}' | tr -d '()')
+              version=${version#*_}
+              if [[ $(printf "%s\n%s" "$version" "$glibcxx_version" | sort -V | tail -n1) == "$version" ]]; then
+                glibcxx_version=$version
+              fi
+            fi
+          done < <("$PWD/.build/sysroots/$TRIPLE/$TRIPLE/bin/objdump" -T "$file")
+
+          if [[ "$glibc_version" != "2.28" ]]; then
+            echo "Error: File $file has dependency on GLIBC > 2.28"
+            exit 1
+          fi
+          if [[ "$glibcxx_version" != "3.4.25" ]]; then
+            echo "Error: File $file has dependency on GLIBCXX > 3.4.25"
+            exit 1
+          fi
+        done
+      condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'))
+      displayName: Check GLIBC and GLIBCXX dependencies in remote/node_modules
```

Comment:
For clients the check will be done during packaging deb and rpm steps, the server was missing this check so added a simple but effective one.